client.go

```diff
@@ -70,29 +89,27 @@ type Client struct {
 }
 
 // NewClient returns a new API Client using the given APIKey
-func NewClient(apiKey string) *Client {
-	return &Client{
-		apiKey:     apiKey,
-		baseURL:    APIHost,
-		Log:        NewLogger(LevelWarn),
-		HTTPClient: defaultClient,
-	}
+func NewClient(apiKey string) (*Client, error) {
+    return NewClientWithOptions(apiKey, nil)
 }
 
-// newClient creates a new Recurly API Client
-func newClient(apiKey string, httpClient *http.Client) *Client {
-	if apiKey == "" {
-		apiKey = APIKey
-	}
-	if httpClient == nil {
-		httpClient = &http.Client{}
-	}
-	return &Client{
+func NewClientWithOptions(apiKey string, options *ClientOptions) (*Client, error) {
```

Comment:
Since this function is being exported from the package, it should have a line above it documentation.

```suggestion
// NewClientWithOptions creates a new client using the given options.
func NewClientWithOptions(apiKey string, options *ClientOptions) (*Client, error) {
```